### PR TITLE
[WEB-1204, WEB-1182] Stat updates

### PR DESCRIPTION
**Pull request by clintonium-119**
See [WEB-1204], [WEB-1182].  Adds `Avg Daily Insulin Ratio` stat to trends view, and sets any of the vertically-tall stats to collapsible.  Stat collapse state is persisted in local storage on a per-view basis.

Related PR: https://github.com/tidepool-org/viz/pull/285

[WEB-1204]: https://tidepool.atlassian.net/browse/WEB-1204?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[WEB-1182]: https://tidepool.atlassian.net/browse/WEB-1182?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test/unit/components/chart/deviceSelection.test.js [hunk 1]
```diff
@@ -160,4 +166,17 @@ describe('DeviceSelection', () => {
     );
     sinon.assert.callCount(props.removeGeneratedPDFS, 4);
   });
+
+  it('should track a metric when device selection accordion is collapsed or expanded', () => {
+    const collapseIcon = wrapper.find('.MuiExpansionPanelSummary-expandIcon').hostNodes();
+    console.log('collapseIcon.debug()', collapseIcon.debug());
```
Comment (krystophv) [inline test/unit/components/chart/deviceSelection.test.js:172]:
```suggestion
```

File: test/unit/components/chart/stats.test.js [hunk 2]
```diff
@@ -106,5 +107,106 @@ describe('Stats', () => {
...
+              // collapsed state of stats
+              averageDailyDose: true,
+              carbs: true,
+            },
+          },
+          sinon.stub()
+        ]));
+
+        wrapper.setProps();
+
```
Comment (krystophv) [inline test/unit/components/chart/stats.test.js:171]: It seems per the comment above that we should verify that `carbStat` is still opened and title hasn't changed even if the localStorage state reflects a collapsed state.

**Review by krystophv — APPROVED** [review]
Code LGTM 👍🏼 

Would suggest adding a couple assertions for one test for completeness.

File: test/unit/components/chart/stats.test.js (hunk 2 above)
Comment (clintonium-119) [reply test/unit/components/chart/stats.test.js:171]: I've added a couple more test assertions to cover this.  Thanks 👍

Comment (krystophv) [reply test/unit/components/chart/stats.test.js:171]: I notice that dailyDoseStat is the return from a function containing a find. Will the `carbStat` reference be properly updated in the new assertions, or would it also need an arrow function to ensure that we're checking the updated value?

Comment (clintonium-119) [reply test/unit/components/chart/stats.test.js:171]: Let me check.

Comment (clintonium-119) [reply test/unit/components/chart/stats.test.js:171]: Yes, good catch.  I've updated that one to a function as well.

**Pull request merged**